iothub/device/src/Transport/DeviceClientEndpointIdentityFactory.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.Azure.Devices.Shared;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Azure.Devices.Client.Transport
+{
+    /// <summary>
+    /// Implementation of the IDeviceClientEndpointIdentityFactory interface
+    /// </summary>
+    internal class DeviceClientEndpointIdentityFactory : IDeviceClientEndpointIdentityFactory
+    {
+        public DeviceClientEndpointIdentity Create(IotHubConnectionString iotHubConnectionString, AmqpTransportSettings amqpTransportSettings, ProductInfo productInfo)
+        {
+            if (Logging.IsEnabled) Logging.Enter(this, $"{nameof(DeviceClientEndpointIdentityFactory)}.{nameof(Create)}");
+
+            DeviceClientEndpointIdentity deviceClientEndpointIdentity;
+
+            if (iotHubConnectionString != null)
+            {
+                if (
+                    (iotHubConnectionString.SharedAccessKey == null) &&
+                    (iotHubConnectionString.SharedAccessKeyName == null) &&
+                    (iotHubConnectionString.SharedAccessSignature == null) &&
+                    (iotHubConnectionString.TokenRefresher == null)
+                   )
+                {
+                    deviceClientEndpointIdentity = new DeviceClientEndpointIdentityX509(iotHubConnectionString, amqpTransportSettings, productInfo);
+                }
+                else if (iotHubConnectionString.SharedAccessKeyName != null)
+                {
+                    deviceClientEndpointIdentity = new DeviceClientEndpointIdentityIoTHubSas(iotHubConnectionString, amqpTransportSettings, productInfo);
+                }
+                else if ((iotHubConnectionString.SharedAccessKeyName == null) || (iotHubConnectionString.TokenRefresher != null))
```

Comment:
at this point (iotHubConnectionString.SharedAccessKeyName == null) should be true always...so do we need the else case below??

Reply:
Fixed - goes with the next check-in

---
In reply to: [251968774](https://github.com/Azure/azure-iot-sdk-csharp/pull/770#discussion_r251968774) [](ancestors = 251968774)